src/Fields/mapreduce.jl

```diff
@@ -67,24 +67,30 @@ end
 """
     norm(v::Field, p=2; normalize=true)
 
-The approximate ``L^p`` norm of `v`. This is computed by summation of the norms
-of the field values multiplied by the Jacobians and the quadrature weights. If
-`normalize=true` (the default), then it is divided by the sum of the weights:
+The approximate ``L^p`` norm of `v`, where ``L^p`` represents the space of measurable
+functions for which the p-th power of the absolute value is Lebesgue integrable.
 ```math
-\\left(\\frac{\\sum_i \\|v_i\\|^p W_i J_i}{\\sum_i W_i J_i}\\right)^{1/p}
+\\| v \\|_p = \\left( \\int_\\Omega |v_i|^p d \\Omega \\right)^{1/p}
+```
+In an `AbstractSpectralElementSpace`, an integral over the entire space is
+computed by summation over the elements of the integrand multiplied by the
+Jacobians and the quadrature weights.
+If `normalize=true` (the default), then it is divided by the sum of the weights:
+```math
+\\left(\\frac{\\sum_i |v_i|^p W_i J_i}{\\sum_i W_i J_i}\\right)^{1/p}
 \\approx
-\\left(\\frac{\\int_\\Omega \\|v\\|^p \\, d \\Omega}{\\int_\\Omega \\, d \\Omega}\\right)^{1/p}
+\\left(\\frac{\\int_\\Omega |v|^p \\, d \\Omega}{\\int_\\Omega \\, d \\Omega}\\right)^{1/p}
```

Comment:
The reason I had `\|` instead of `|` is that for non-scalar-valued fields we call `norm` at each point. This should probably be explained a bit better.

Reply:
It looks odd to have a norm inside of a norm definition. It looks circular.... Should we just add a subscript to denote the component then? Like 
```
\\int_\\Omega |v_i|^p \\, d \\Omega
```
In case of a scalar-valued function then `v_i \equiv v`

Reply:
its a recursive norm (like Base julia), what Simon had looks like it aligns more with how norm is documented in `LinearAlgebra.norm`

Reply:
See https://docs.julialang.org/en/v1/stdlib/LinearAlgebra/#LinearAlgebra.norm

Reply:
From what I see in the [`LinearAlgebra.norm` docs](https://docs.julialang.org/en/v1/stdlib/LinearAlgebra/#LinearAlgebra.norm) they define the l^2 norm (small l, not capital L. Suitable for discrete [sequence spaces](https://en.wikipedia.org/wiki/Lp_space#General_%E2%84%93p-space), not L^2 continuous function spaces like what we are trying to define here) and they also have absolute values of each component in the definition

Reply:
I'm happy to switch notation so that the inner norm uses |v|, but we should explain that this will call `norm`, as per the link:
> ∣a_i∣| the [norm](https://docs.julialang.org/en/v1/stdlib/LinearAlgebra/#LinearAlgebra.norm) of a_i.

> Since the p-[norm](https://docs.julialang.org/en/v1/stdlib/LinearAlgebra/#LinearAlgebra.norm) is computed using the norms of the entries of A, the p-norm of a vector of vectors is not compatible with the interpretation of it as a block vector in general if p != 2.

Reply:
I added a clarification about vector-valued Vs scalar Fields in [[b72dfb9](https://github.com/CliMA/ClimaCore.jl/pull/507/commits/b72dfb965707503948e78fbe0a244a02d51ba3a6)](https://github.com/CliMA/ClimaCore.jl/pull/507/commits/b72dfb965707503948e78fbe0a244a02d51ba3a6). If we want to be more precise, we should use a double index. One for the Field component (I would personally call that `c`, unlike what we have used for the `IJKFVH` DataLayout) and one for the nodes

Reply:
A recursive norm is calling calling `norm_p(vector_val)` it's not component wise?   Maybe I'm confused about how it is worded.